dnf5/commands/list/list.cpp

```diff
@@ -0,0 +1,284 @@
+    auto sections = create_output();
+
+    switch (pkg_narrow) {
+        case PkgNarrow::ALL: {
+            libdnf::rpm::PackageQuery installed(base_query);
+            installed.filter_installed();
+            libdnf::rpm::PackageQuery available(base_query);
+            available.filter_available();
+            if (!showduplicates) {
+                available.filter_priority();
+                available.filter_latest_evr();
+                // keep only those available packages that are either not installed or
+                // available EVR is higher than the installed one
+                libdnf::rpm::PackageQuery installed_latest(installed);
+                installed_latest.filter_latest_evr();
+                available.filter_nevra(installed_latest, libdnf::sack::QueryCmp::NOT | libdnf::sack::QueryCmp::LTE);
+            }
+            package_matched |= sections->add_section("Installed packages", installed);
+            {
+                // TODO(mblaha) currently only the installed version and upgrades
+                // are highlighted to make the output a bit saner
+                auto colorizer = std::make_unique<libdnf::cli::output::PkgColorizer>(
+                    installed,
+                    "",  //config.color_list_available_install().get_value(),
+                    "",  //config.color_list_available_downgrade().get_value(),
+                    config.color_list_available_reinstall().get_value(),
+                    config.color_list_available_upgrade().get_value());
+                package_matched |= sections->add_section("Available packages", available, colorizer);
+            }
+            break;
+        }
+        case PkgNarrow::INSTALLED: {
+            base_query.filter_installed();
+            package_matched |= sections->add_section("Installed packages", base_query);
+            break;
+        }
+        case PkgNarrow::AVAILABLE: {
+            libdnf::rpm::PackageQuery installed(base_query);
+            installed.filter_installed();
+            base_query.filter_available();
+            if (!showduplicates) {
+                base_query.filter_priority();
+                base_query.filter_latest_evr();
+            }
+            {
+                // TODO(mblaha) currently only the installed version and upgrades
+                // are highlighted to make the output a bit saner
+                auto colorizer = std::make_unique<libdnf::cli::output::PkgColorizer>(
+                    installed,
+                    "",  //config.color_list_available_install().get_value(),
+                    "",  //config.color_list_available_downgrade().get_value(),
+                    config.color_list_available_reinstall().get_value(),
+                    config.color_list_available_upgrade().get_value());
+                package_matched |= sections->add_section("Available packages", base_query, colorizer);
+            }
+            break;
+        }
+        case PkgNarrow::UPGRADES:
```

Comment:
I am not sure if this is intended but using `dnf5 list pkg --upgrades` also includes src packages.
This is a change when compared to dnf4.

Reply:
Thanks, it's a mistake. I'll fix it.